### PR TITLE
Added echo state network (ESN)

### PR DESCRIPTION
**Pull request by pedrolarben**
Hello, 
I have been working with recurrent neural networks. I needed an implementation of Echo State Networks (ESN) for tensorflow. I found some implementations, but none of them was compatible with tensorflow2, so I implemented it. I think it would be great to add ESN to tensorflow_addons so everybody can use it. 

> "Echo state networks (ESN) provide an architecture and supervised learning principle for recurrent neural networks (RNNs). The main idea is (i) to drive a random, large, fixed recurrent neural network with the input signal, thereby inducing in each neuron within this "reservoir" network a nonlinear response signal, and (ii) combine a desired output signal by a trainable linear combination of all of these response signals". 
>Herbert Jaeger, Jacobs University Bremen, Bremen, Germany 

Echo state network scheme by Peter Birkholz<br><br>

**References**

- The “echo state” approach to analysing and training recurrent neural networks-with an erratum note. [Herbert Jaeger](https://scholar.google.de/citations?user=0uztVbMAAAAJ&hl=en), 2001. Cited by 2281. [[pdf]](https://www.researchgate.net/profile/Herbert_Jaeger3/publication/215385037_The_echo_state_approach_to_analysing_and_training_recurrent_neural_networks-with_an_erratum_note'/links/566a003508ae62b05f027be3/The-echo-state-approach-to-analysing-and-training-recurrent-neural-networks-with-an-erratum-note.pdf)
- [ESN implementation for tensorflow 1.12](https://github.com/m-colombo/Tensorflow-EchoStateNetwork)
- ESN implementations for pytorch: [EchoTorch](https://github.com/nschaetti/EchoTorch), [pytorch-esn](https://github.com/stefanonardo/pytorch-esn)

### CODE REVIEW

**Comment by pedrolarben** [comment]
@googlebot I signed it!

File: tensorflow_addons/layers/esn.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        activation: Activation = "tanh",
+        kernel_initializer: Initializer = "glorot_uniform",
+        recurrent_initializer: Initializer = "glorot_uniform",
+        bias_initializer: Initializer = "zeros",
+        return_sequences=False,
+        go_backwards=False,
+        unroll=False,
+        **kwargs
+    ):
+        cell = tfa.rnn.cell.ESNCell(
```
Comment (pedrolarben) [inline tensorflow_addons/layers/esn.py:111]: Automatic test with Bazel is failing here. 
The error says ` AttributeError: module 'tensorflow_addons' has no attribute 'rnn'`. However, tests pass locally.
I may be failing to import the dependencies. 
Any clue on how to fix it?

Comment (pedrolarben) [reply tensorflow_addons/layers/esn.py]: My fault. Fixed!

**Comment by gabrieldemarmiesse** [comment]
Thank you for the pull request! I think we need to split the pull request in two to make it easier to review. Could you make a first pull request with only the cell? Then you can add the layer later on.

**Comment by pedrolarben** [comment]
@gabrieldemarmiesse sure, here it is the pull request with the ESNCell https://github.com/tensorflow/addons/pull/1811 

I close this pull request.

**Comment by pedrolarben** [comment]
@gabrieldemarmiesse The first part of this PR (the rnn cell) has been approved https://github.com/tensorflow/addons/pull/1811, if someone with write access could merge it to the repository, I will proceed with the remaining code (the layer). Thank you very much.

**Comment by gabrieldemarmiesse** [comment]
@pedrolarben the PR is merged :)

**Pull request closed without merging**